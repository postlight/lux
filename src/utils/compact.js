/* @flow */

import { isNil } from '@lux/utils/is-type'

const compact = <T>(source: Array<T>): Array<T> =>
  source.filter(value => !isNil(value))

export default compact
