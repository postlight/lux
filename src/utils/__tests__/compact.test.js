/* @flow */

import compact from '../compact'

test('compact()', () => {
  expect(compact([0, 'a', 1, null, {}, undefined, false])).toMatchSnapshot()
})
