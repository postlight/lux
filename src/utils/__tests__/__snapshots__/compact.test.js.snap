// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compact() 1`] = `
Array [
  0,
  "a",
  1,
  Object {},
  false,
]
`;
